### PR TITLE
Proper systemerror follow-up

### PR DESCRIPTION
**Pull request by maurolacy**
Adds a custom `ErrNoSuchContract` error, and map it to the corresponding `wasmvmtypes` error, as discussed in #701.

Needs feedback on registering the error, proper error names / types, and extending this mechanism to other errors.

### CODE REVIEW

File: x/wasm/types/errors.go [hunk 1]
```diff
@@ -69,4 +69,23 @@ var (
 
 	// ErrInvalidEvent error if an attribute/event from the contract is invalid
 	ErrInvalidEvent = sdkErrors.Register(DefaultCodespace, 21, "invalid event")
+
+	// ErrNoSuchContract error if an address does not belong to a contract
```
Comment (ethanfrey) [inline x/wasm/types/errors.go:73]: Why do we need this if we have the custom type? Just to ensure no one else uses it?

If so, I would name it `errNoSuchContract` or such to make it private, just register it

**Review by ethanfrey — COMMENTED** [review]
nice. a few tips.

I will let Alex approve

File: x/wasm/keeper/query_plugins.go [hunk 2]
```diff
@@ -56,7 +56,18 @@ func (q QueryHandler) Query(request wasmvmtypes.QueryRequest, gasLimit uint64) (
 	defer func() {
 		q.Ctx.GasMeter().ConsumeGas(subCtx.GasMeter().GasConsumed(), "contract sub-query")
 	}()
-	return q.Plugins.HandleQuery(subCtx, q.Caller, request)
+
+	res, err := q.Plugins.HandleQuery(subCtx, q.Caller, request)
+	// Error mapping
+	if err != nil {
+		switch err.(type) {
```
[the remark this answers is not in the document]
Comment (maurolacy) [reply x/wasm/keeper/query_plugins.go:63]: I think we expect `*types.NoSuchContract` here. Will change to the typical idiom.

**Comment by alpe** [comment]
I am discussing with @maurolacy how to make this work with [wasmvm systemerror](https://github.com/CosmWasm/wasmvm/blob/main/types/systemerror.go#L93) without circular dependency.

**Comment by ethanfrey** [comment]
> I am discussing with @maurolacy how to make this work with [wasmvm systemerror](https://github.com/CosmWasm/wasmvm/blob/main/types/systemerror.go#L93) without circular dependency.

Right now he converts the standard type to the expected wasmvmtype so that `ToSystemError` works properly.
I think any other approach that would make `ToSystemError` conscious of this custom type and type conversion would require at least as much code and likely be more fragile.

The only thing I could think of to make this more extensible (if we do need to use this multiple places in the codebase) is to pull out [these lines](https://github.com/CosmWasm/wasmd/pull/715/files#diff-31a3ad60ef3595b573c7ea5150e6bb8176b41e261ba7faa5a5e7769cc4a7b928R62-R65) into a `MapErr` function.

**Pull request merged**